arrow-fx/src/test/kotlin/arrow/fx/MVarTest.kt

```diff
@@ -181,13 +181,13 @@ class MVarTest : UnitSpec() {
               ch.put(1).flatMap { loop(n - 1, acc + x, ch) }
             }
 
-        val count = 10000
+        val count = 100
```

Comment:
The test is for stack safety. If this blows up then there's something wrong in the underlying implementation.

Reply:
Yes, exactly. I temporarily reduced the counters in order to see if there were any other problem or test failing. Before proceeding with this PR all the impacted tests/code should be fixed. For the Rx2 problem I made a fix, but I'm not sure it is the best one. For the other ones, I haven't analysed them yet.